app/components/contact/contact-item/contact-item-modal.js

```diff
@@ -0,0 +1,50 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { isBlank } from '@ember/utils';
+
+export default class ContactContactItemModalComponent extends Component {
+  @service currentSession;
+  @service store;
+
+  @tracked contact;
+
+  constructor() {
+    super(...arguments);
+    if (this.args.contact) {
+      this.contact = this.args.contact;
+    } else {
+      const creator = this.currentSession.organization;
+      const telephone = this.store.createRecord('telephone', {creator});
+      const mailAddress = this.store.createRecord('mail-address', {creator});
+      this.contact = this.store.createRecord('contact-item', {telephone, mailAddress});
```

Comment:
Het aanmaken van het nieuwe record zou ik buiten deze component houden en in de `openContactItemModal` action van de `new.js` controller doen. Op die manier wordt deze component eenvoudiger: hij krijgt in alle gevallen een contact binnen in `@contact`. De lokale `this.contact` lijkt me dan ook niet meer nodig.